test/integration/ColdResumptionTest.cpp

```diff
@@ -0,0 +1,192 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include <chrono>
+#include <condition_variable>
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+
+#include <folly/io/async/ScopedEventBaseThread.h>
+
+#include "test/integration/ClientUtils.h"
+#include "test/integration/ServerFixture.h"
+
+using namespace std::chrono_literals;
+using namespace ::reactivesocket;
+using namespace ::testing;
+
+using folly::ScopedEventBaseThread;
+
+// A very simple test which tests a basic cold resumption workflow.
+// This setup can be used to test varying scenarious in cold resumption.
+TEST_F(ServerFixture, ColdResumptionBasic) {
+  ScopedEventBaseThread eventBaseThread;
+  auto clientEvb = eventBaseThread.getEventBase();
+  tests::MyConnectCallback connectCb;
+  auto token = ResumeIdentificationToken::generateNew();
+  std::unique_ptr<ReactiveSocket> rsocket;
+  auto mySub = std::make_shared<tests::MySubscriber>();
+  auto resumeCache = std::make_shared<ResumeCache>();
+  Sequence s;
+  SCOPE_EXIT {
+    clientEvb->runInEventBaseThreadAndWait([&]() { rsocket.reset(); });
+  };
+
+  // The thread running this test, and the thread running the clientEvb
+  // have to synchronized with a barrier.
+  std::mutex cvM;
+  std::condition_variable cv;
+  std::unique_lock<std::mutex> lk(cvM);
+
+  // Get a few subscriptions (happens right after connecting)
+  EXPECT_CALL(*mySub, onSubscribe_()).WillOnce(Invoke([&]() {
+    mySub->request(3);
+  }));
+  EXPECT_CALL(*mySub, onNext_("1"));
+  EXPECT_CALL(*mySub, onNext_("2"));
+  EXPECT_CALL(*mySub, onNext_("3")).WillOnce(Invoke([&](std::string) {
```

Comment:
should we expect call to onComplete or onError since we close the RS instance with this subscriber?